src/tests/eventpipe/GCEvents.cs

```diff
@@ -206,26 +206,27 @@ await RemoteTestExecutorHelper.RunTestCaseAsync(() => {
                 Func<EventPipeEventSource, Func<int>> _DoesTraceContainEvents = (source) => {
                     int GCCreateSegmentEvents = 0;
                     int GCFreeSegmentEvents = 0;
-                    source.Clr.GCCreateSegment += (eventData) => GCCreateSegmentEvents += 1;
-                    source.Clr.GCFreeSegment += (eventData) => GCFreeSegmentEvents += 1;
+                    if (Environment.Version.Major < 7)
+                    {
+                        source.Clr.GCCreateSegment += (eventData) => GCCreateSegmentEvents += 1;
+                        source.Clr.GCFreeSegment += (eventData) => GCFreeSegmentEvents += 1;
+                    }
 
                     int GCAllocationTickEvents = 0;
                     source.Clr.GCAllocationTick += (eventData) => GCAllocationTickEvents += 1;
 
-                    int GCCreateConcurrentThreadEvents = 0;
-                    int GCTerminateConcurrentThreadEvents = 0;
-                    source.Clr.GCCreateConcurrentThread += (eventData) => GCCreateConcurrentThreadEvents += 1;
-                    source.Clr.GCTerminateConcurrentThread += (eventData) => GCTerminateConcurrentThreadEvents += 1;
-
                     return () => {
                         Logger.logger.Log("Event counts validation");
 
-                        Logger.logger.Log("GCCreateSegmentEvents: " + GCCreateSegmentEvents);
-                        Logger.logger.Log("GCFreeSegmentEvents: " + GCFreeSegmentEvents);
+                        bool GCSegmentResult = true;
+                        if (Environment.Version.Major < 7)
+                        {
+                            Logger.logger.Log("GCCreateSegmentEvents: " + GCCreateSegmentEvents);
+                            Logger.logger.Log("GCFreeSegmentEvents: " + GCFreeSegmentEvents);
 
-                        // Disable checking GCFreeSegmentEvents on .NET 7.0 issue: https://github.com/dotnet/diagnostics/issues/3143
-                        bool GCSegmentResult = GCCreateSegmentEvents > 0 && (GCFreeSegmentEvents > 0 || Environment.Version.Major >= 7);
-                        Logger.logger.Log("GCSegmentResult: " + GCSegmentResult);
+                            GCSegmentResult = GCCreateSegmentEvents > 0 && GCFreeSegmentEvents > 0;
```

Comment:
Do we also expect the GCCreateSegmentEvents not to be generated? The test appears to have stopped checking both but the comments only mention GCFreeSegment events stopped. Assuming CreateSegment events are still expected I'd suggest only removing the GCFreeSegmentEvents portion of the check.

I don't think we need to make it conditional on major version either. As long as the test won't start failing when run on older versions it should be fine.

Reply:
In the runtime, there are still technically instances of firing `GCCreateSegment_V1` and `GCFreeSegment_V1` events. However, from offline discussions summarized in https://github.com/dotnet/diagnostics/issues/3143#issuecomment-2330179117, this PR is driving towards removing expectations of those events being fired during the new GC.

Since both GCCreateSegment and GCFreeSegment are documented to specifically mean GC segments being created/released, and since the new GC uses regions, I figured that it would be better keep those events to specifically mean GC segments.

Reply:
if both of those events are affected thats np, the comments elsewhere had given a false impression that the FreeSegment events were the only one impacted.